server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -564,6 +571,15 @@ IndexShardStats indexShardStats(final IndicesService indicesService, final Index
         );
     }
 
+    private Map<Index, ProjectId> projectsByIndex() {
+        Map<Index, ProjectId> map = new HashMap<>();
+        for (IndexService indexShards : indices.values()) {
+            Index index = indexShards.index();
+            clusterService.state().metadata().lookupProject(index).ifPresent(project -> map.put(index, project.id()));
```

Comment:
Do we want to assert and/or log a warning if the project is not present?

Reply:
Ah nvm, I see you're explaining this choice in your other comment. Ignore this comment please.